tests/e2e/lib/restic_helpers.go

```diff
@@ -47,64 +47,57 @@ func HasCorrectNumResticPods(namespace string) wait.ConditionFunc {
 	}
 }
 
-func waitForDesiredResticPods(namespace string) error {
-	return wait.PollImmediate(time.Second*5, time.Minute*2, HasCorrectNumResticPods(namespace))
-}
+var lastObservedReadyResticDaemeonSet *appsv1.DaemonSet
 
-func AreResticPodsRunning(namespace string) wait.ConditionFunc {
-	log.Printf("Checking for correct number of running Restic pods...")
+func AreResticDaemonsetUpdatedAndReady(namespace string) wait.ConditionFunc {
+	log.Printf("Checking if Restic daemonset is ready...")
 	return func() (bool, error) {
-		err := waitForDesiredResticPods(namespace)
-		if err != nil {
-			return false, err
-		}
-		client, err := setUpClient()
+		rds, err := GetResticDaemonSet(namespace, "restic")
 		if err != nil {
 			return false, err
 		}
-		resticPodOptions := metav1.ListOptions{
-			LabelSelector: "name=restic",
-		}
-		// get pods in the oadp-operator-e2e namespace with label selector
-		podList, err := client.CoreV1().Pods(namespace).List(context.TODO(), resticPodOptions)
-		if err != nil {
+		if rds.Status.UpdatedNumberScheduled != rds.Status.DesiredNumberScheduled ||
+			rds.Status.NumberUnavailable != 0 {
+			log.Printf("Restic daemonset is not ready with condition: %v", rds.Status.Conditions)
 			return false, nil
 		}
-		// loop until pod status is 'Running' or timeout
-		for _, podInfo := range (*podList).Items {
-			if podInfo.Status.Phase != "Running" {
-				return false, err
-			}
+		if lastObservedReadyResticDaemeonSet != nil &&
+			lastObservedReadyResticDaemeonSet.CreationTimestamp.Equal(&rds.CreationTimestamp) && // same object
+			!reflect.DeepEqual(lastObservedReadyResticDaemeonSet.Spec, rds.Spec) &&
+			// if spec is not deep equal, something updated. Consider ready only if observedGeneration has increased.
+			rds.Status.ObservedGeneration == lastObservedReadyResticDaemeonSet.Status.ObservedGeneration {
```

Comment:
DeepEqual check seems redundant here. it would make more sense to deepequal an updated spec passing in as an optional parameter.